### PR TITLE
1.2

### PR DESCRIPTION
**Pull request by ollakalla**
I have make better code for handling Multi-levels of arrays , also i have add the test case for my code
supposed everything now goes right :)

### CODE REVIEW

File: src/Xml/Document.php [hunk 1]
```diff
@@ -196,34 +199,247 @@ protected function getValueCollection(SimpleXMLElement $content, array $matches,
...
+                $value = array_merge($value, $item);
+            } else {
+                Arr::set($value, $as, $item);
+            }
+        } else {
+            if ($name == '@') {
+                $name = null;
+            }
+            Arr::set($value, $as, $this->getValue($content, $name));
+        }
```
Comment (crynobone) [inline src/Xml/Document.php]: I need some tests that cover this use case.

https://coveralls.io/builds/16556955/source?filename=src%2FXml%2FDocument.php

File: src/Xml/Document.php [hunk 2]
```diff
@@ -196,34 +199,247 @@ protected function getValueCollection(SimpleXMLElement $content, array $matches,
...
+    }
+
+    /**
+     * split the use .
+     *
+     * @param  string  $string
+     *
+     * @return array
+     */
+    public function specialSplitAdvanced($string) {
```
Comment (crynobone) [inline src/Xml/Document.php]: I don't think this need to be `public`.

File: src/Xml/Document.php [hunk 3]
```diff
@@ -196,34 +199,247 @@ protected function getValueCollection(SimpleXMLElement $content, array $matches,
...
+    }
+
+    /**
+     * split the use .
+     *
+     * @param  string  $string
+     *
+     * @return array
+     */
+    public function specialSplit($string) {
```
Comment (crynobone) [inline src/Xml/Document.php]: Shouldn't be `public`.

File: src/Xml/Document.php [hunk 4]
```diff
@@ -196,34 +199,247 @@ protected function getValueCollection(SimpleXMLElement $content, array $matches,
...
+                                $as = null;
+                            }
+
+                            $item = $this->getSelfMatchingValue($content, $matches, $as);
+
+                            if (is_null($as)) {
+                                $value = array_merge($value, $item);
+                            } else {
+                                Arr::set($value, $as, $item);
+                            }
```
Comment (crynobone) [inline src/Xml/Document.php]: This clause doesn't have any test. Need to at least make sure the `preg_match` can validate expected values.

https://coveralls.io/builds/16556955/source?filename=src%2FXml%2FDocument.php

File: src/Xml/Document.php (hunk 2 above)
Comment (ollakalla) [reply src/Xml/Document.php]: Ok no problem

File: src/Xml/Document.php (hunk 3 above)
Comment (ollakalla) [reply src/Xml/Document.php]: Ok no problem

File: src/Xml/Document.php (hunk 4 above)
Comment (ollakalla) [reply src/Xml/Document.php]: actually this code is already exist, i don't add it. 
i just handle the case of multi levels of array, and keep the existing code,

File: src/Xml/Document.php (hunk 1 above)
Comment (ollakalla) [reply src/Xml/Document.php]: Sorry i forget to remove this function, it is just for test something, i will remove it

File: src/Xml/Document.php (hunk 4 above)
Comment (ollakalla) [reply src/Xml/Document.php]: Ok there was bug in the code, i have fixed it and add the test case for this .
i will push when i finish :)

Comment (crynobone) [reply src/Xml/Document.php]: The code was originally from https://github.com/orchestral/parser/issues/7 which introduce https://github.com/orchestral/parser/commit/82c29a7364749ff092dab021b7e4dc878ef7731a

screen shot 2018-04-18 at 6 02 19 pm

Comment (ollakalla) [reply src/Xml/Document.php]: i get it .
i have finished updating code . and pushed it.
but i can't see it on the pull request

Comment (crynobone) [reply src/Xml/Document.php]: You can generate the coverage locally by running the following command.

```
./vendor/bin/phpunit --coverage-html=coverage/
```

and the `open coverage/index.html` on your browser (depending on your OS).

Comment (ollakalla) [reply src/Xml/Document.php]: i have generate coverage locally and add the test case for this code, every thing is good right now :+1:

**Pull request merged**